src/datanode/src/datanode.rs

```diff
@@ -346,11 +345,22 @@ impl DatanodeBuilder {
         while let Some(table_value) = table_values.next().await {
             let table_value = table_value.context(GetMetadataSnafu)?;
             for region_number in table_value.regions {
+                // Augments region options with wal options.
+                // TODO(niebayes): fetch wal options map from region info.
+                // let wal_options_map = &table_value.region_info.wal_options_map;
+                let wal_options_map: HashMap<RegionNumber, EncodedWalOptions> = HashMap::default();
+                let wal_options = wal_options_map
+                    .get(&region_number)
+                    .cloned()
+                    .unwrap_or_default();
+                let mut region_options = table_value.region_info.region_options.clone();
+                region_options.extend(wal_options);
```

Comment:
Can we pass in `wal_options` directly instead of put it in `region_options`? `extend` might implicitly clone the vec.

Reply:
The `wal_options` is now encoded as a string and hence no need to extend.